core/src/main/java/hudson/util/RingBufferLogHandler.java

```diff
@@ -66,16 +67,16 @@ public static int getDefaultRingBufferSize() {
     @Override
     public synchronized void publish(LogRecord record) {
         int len = records.length;
-        records[(start+size)%len]=record;
-        if(size==len) {
+        records[(start+size.get())%len]=record;
+        if(size.get()==len) {
```

Comment:
Bonus points for using a local variable. It is synchronized anyway

Reply:
You are right 👍 and done